pkg/agent/clientset.go

```diff
@@ -183,9 +183,13 @@ func (cs *ClientSet) sync() {
 	for {
 		if err := cs.connectOnce(); err != nil {
 			if dse, ok := err.(*DuplicateServerError); ok {
-				klog.V(4).InfoS("duplicate server", "serverID", dse.ServerID, "serverCount", cs.serverCount, "clientsCount", cs.ClientsCount())
-				if cs.serverCount != 0 && cs.ClientsCount() >= cs.serverCount {
+				clientsCount := cs.ClientsCount()
+				klog.V(4).InfoS("duplicate server", "serverID", dse.ServerID, "serverCount", cs.serverCount, "clientsCount", clientsCount)
+				if cs.serverCount != 0 && clientsCount >= cs.serverCount {
 					duration = backoff.Step()
+				} else if clientsCount < cs.serverCount {
```

Comment:
Note there is a 3rd implicit else case where we still do nothing. What do you think of this simplification:

```
if ... { # safe to go slower
  duration = backoff.Step()
} else ... { # should switch back to faster
  backoff = ...
  duration = ...
}
```

Reply:
Sounds good to me. Changed it to:
```go
if (safe to go slower) {
    duration = backoff.Step()
} else {
    // reset backoff
}
```